libs/vm-modules/tests/unit/vm_charge_tests.cpp

```diff
@@ -463,6 +482,23 @@ TEST_F(
   ASSERT_TRUE(toolkit.Run(nullptr, high_charge_limit));
 }
 
+TEST_F(VmChargeTests,
+       function_bind_with_charge_estimate_execution_fails_when_charge_would_overflow_with_estimator)
+{
+  toolkit.module()
+      .CreateClassType<CustomType>("CustomType")
+      .CreateMemberFunction("overflowExpensive", &CustomType::OverflowExpensive, max_charge_amount);
+
+  static char const *TEXT = R"(
+    function main()
+      CustomType.overflowExpensive(3u8, 4u16);
+    endfunction
+  )";
+
+  ASSERT_TRUE(toolkit.Compile(TEXT));
+  ASSERT_FALSE(toolkit.Run(nullptr, max_charge_amount));
+}
+
```

Comment:
We don't need the complications of member functions, we're just writing an overflow check. Replace this second test with the following (nearly identical to the first test, but note use of `max_charge_estimator` instead of `max_charge_amount`).

```
TEST_F(VmChargeTests, functor_bind_with_charge_estimate_execution_does_not_overflow_charge_total_with_estimator)
{
  toolkit.module().CreateFreeFunction("overflowExpensive", handler, max_charge_estimator);

  static char const *TEXT = R"(
    function main()
      overflowExpensive(3u8, 4u16);
    endfunction
  )";

  ASSERT_TRUE(toolkit.Compile(TEXT));
  ASSERT_FALSE(toolkit.Run(nullptr, max_charge_amount));
}
```